internal/controller/bindings/forwarder_controller.go

```diff
@@ -112,11 +120,112 @@ func (r *ForwarderReconciler) update(ctx context.Context, epb *bindingsv1alpha1.
 		"port", epb.Spec.Port,
 	)
 
+	// Get the KubernetesOperator
+
+	operators := ngrokv1alpha1.KubernetesOperatorList{}
+	if err := r.Client.List(ctx, &operators); err != nil {
+		log.Error(err, "failed to list KubernetesOperators")
+		return err
+	}
+	if len(operators.Items) == 0 {
+		log.Info("no KubernetesOperators found")
+		return fmt.Errorf("no KubernetesOperators found")
+	}
+	op := operators.Items[0]
+
+	// Bindings should be enabled on the operator, if they aren't we can't do anything
+	if op.Spec.Binding == nil {
+		log.Info("ut oh, no binding", "op", op)
+		return fmt.Errorf("no binding")
+	}
+
+	if op.Spec.Binding.IngressEndpoint == nil {
+		log.Info("ut oh, no ingress endpoint", "op", op)
+		return fmt.Errorf("no ingress endpoint")
+	}
+
+	// Get the secret
+	secret := v1.Secret{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: op.Namespace, Name: op.Spec.Binding.TlsSecretName}, &secret); err != nil {
+		return err
+	}
+
+	keyData, hasKey := secret.Data["tls.key"]
+	certData, hasCert := secret.Data["tls.crt"]
+
+	if !hasKey || !hasCert {
+		return fmt.Errorf("missing tls.key or tls.crt")
+	}
+
+	cert, err := tls.X509KeyPair(certData, keyData)
+	if err != nil {
+		return err
+	}
+
+	tlsDialer := tls.Dialer{
+		NetDialer: &net.Dialer{
+			Timeout: 3 * time.Minute,
+		},
+		Config: &tls.Config{
+			Certificates: []tls.Certificate{cert},
+		},
+	}
+
+	endpointURI, err := url.Parse(epb.Spec.EndpointURI)
+	if err != nil {
+		return err
+	}
+
+	host := endpointURI.Hostname()
+	port, err := strconv.Atoi(endpointURI.Port())
```

Comment:
OOC: Why `Atoi` and then later `Itoa` for the port?

Reply:
Just what I had for local variables. I'll switch it to capture an intermediate string var for the port

Reply:
Fixed. Removed the Itoa below